src/services/healthcareProfessionalService.ts

```diff
@@ -17,9 +23,24 @@ export const getHealthcareProfessionalById = async (id: string) : Promise<Health
     return convertedEntity
 }
 
-export const addHealthcareProfessional = async (healthcareProfessional : HealthcareProfessional) : Promise<void> => {
+export const addHealthcareProfessional = async (input : HealthcareProfessionalInput) : Promise<string[]> => {
+    const db = getFirestore()
+   
+    const healthcareProRef = db.collection('healthcareProfessionals')
+
+    const newHealthcareProfessional = {
+        acceptedInsurance: mapAndValidateInsurance(input.acceptedInsurance as []),
+        degrees: mapAndValidateDegrees(input.degrees as DegreeInput[]),
+        names: mapAndValidateNames(input.names as LocaleNameInput[]),
+        specialties: mapAndValidateSpecialties(input.specialties as SpecialtyInput[]),
+        spokenLanguages: mapAndValidateLanguages(input.spokenLanguages as SpokenLanguageInput[])
+    }
+
+    const idList: string[] = []
+    
+    await healthcareProRef.add(newHealthcareProfessional).then((docRef: DocumentReference) => idList.push(docRef.id))
```

Comment:
don't use `.then()` with `await`. They're conflicting designs. Try to use async/await where possible